packages/dev/core/src/Shaders/gaussianSplatting.vertex.fx

```diff
@@ -0,0 +1,76 @@
+﻿precision highp float;
+
+#include<helperFunctions>
+
+// Attributes
+attribute vec3 position;
+//#include<instancesDeclaration>

```

Comment:
I could not get the define in instancesDeclaration to be true. What am I missing here?

Reply:
The mesh being an `InstancedMesh` or having thin instances defined should be enough to set the `INSTANCES` define.

However, when I did this test (https://playground.babylonjs.com/#6R85ZB#46), I added a wait (see `checkReady`) to be sure to start rendering the GS only when thin instances have been set on the mesh. Else, the shader code would fail because it accesses `world0`/../`world3` attributes, which don't exist until thin instance matrices have been set.

Maybe you have the same problem?

Reply:
I think the easiest way would be to create a single thin instance just after the mesh is created (with a (0,0,0) scale, for eg, so that nothing is rendered - or a translation that clips it entirely). This thin instance will be overwritten by the `thinInstanceSetBuffer` / `thinInstanceBufferUpdated` call later on.